File: 3016-minimum-number-of-pushes-to-type-word-ii/3016-minimum-number-of-pushes-to-type-word-ii.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        freq = [0] * 26
        
        for letter in word:
            freq[ord(letter) - 97] += 1
        
        freq.sort(reverse=True)
        
        ans = 0
        for i in range(len(freq)):
            if freq[i] == 0: 
                break
            check = (i // 8) + 1
            ans += freq[i] * check
        return ans

File: 3016-minimum-number-of-pushes-to-type-word-ii/3016-minimum-number-of-pushes-to-type-word-ii.js
/**
 * @param {string} word
 * @return {number}
 */
var minimumPushes = function(word) {
    let freq = new Array(26).fill(0)
    
    for(let i = 0; i < word.length; i++){
        freq[word.charCodeAt(i)-97]++
    }
    freq.sort((a,b) => b - a)
    
    let ans = 0;
    for(let i = 0; i < freq.length; i++){
        if(freq[i] == 0) break;
        let check = Math.floor(i / 8)+1;

        ans += freq[i] * check
    }
    return ans
};